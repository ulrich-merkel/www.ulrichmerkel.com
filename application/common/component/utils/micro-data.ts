/**
 * Es6 module for creating mirco data attributes.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isValidString } from '../../utils/string';

type ItemTypeAttributes = {
    itemScope: boolean;
    itemType: string;
};

/**
 * Get component properties for itemProp and itemType.
 *
 * @param {boolean} itemType - The itemType property if available
 * @returns {object|null} The component props for spreading
 */
export function getItemTypeAttributes(
    itemType?: string
): ItemTypeAttributes | null {
    return isValidString(itemType)
        ? {
              itemScope: true,
              itemType
          }
        : null;
}
