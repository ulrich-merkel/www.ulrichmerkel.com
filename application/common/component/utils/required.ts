/**
 * Es6 module for creating form input required attributes.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isBoolean } from 'lodash';

type RequiredAttributes = {
    'aria-required': boolean;
    required: boolean;
};

/**
 * Get component properties for required form input elements.
 *
 * @param {boolean} required - Whether the input element is required or not
 * @returns {object|null} The component props for spreading
 */
export function getRequiredAttributes(
    required?: boolean
): RequiredAttributes | null {
    return isBoolean(required) && required
        ? {
              'aria-required': true,
              required: true
          }
        : null;
}
