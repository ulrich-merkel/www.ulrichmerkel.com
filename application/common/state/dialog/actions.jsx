/* eslint-disable import/prefer-default-export */
/**
 * Es6 module for Redux Architecture.
 * Actions are payloads of information that send data from your application
 * to your store. They are the only source of information for the store. You
 * send them to the store using store.dispatch().
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2016
 * @version 0.0.1
 *
 * @requires common/state/dialog/constants
 *
 * @changelog
 * - 0.0.1 Basic functions and structure
 */
// @TODO Rename STATE_DIALOG_PAGE_BROADCAST to STATE_DIALOG_CONTENT_BROADCAST
import {
    STATE_DIALOG_VISIBLE_CHANGE,
    STATE_DIALOG_PAGE_BROADCAST,
    STATE_DIALOG_PAGE_SEARCH,
    STATE_DIALOG_PAGE_THEME
} from './constants';

/**
 * Handle dialog visibility state change.
 *
 * @function
 * @param {boolean} visible - The new visible state
 * @returns {object} The redux action playload
 */
function changeDialogVisible(visible) {
    return {
        type: STATE_DIALOG_VISIBLE_CHANGE,
        visible
    };
}

/**
 * Handle dialog visibility state change.
 *
 * @function
 * @param {boolean} visible - The new visible state
 * @returns {object} The redux action playload
 */
function changeDialogVisibleBroadcast(visible) {
    return {
        type: STATE_DIALOG_VISIBLE_CHANGE,
        page: STATE_DIALOG_PAGE_BROADCAST,
        visible
    };
}

/**
 * Handle dialog visibility state change.
 *
 * @function
 * @param {boolean} visible - The new visible state
 * @returns {object} The redux action playload
 */
function changeDialogVisibleSearch(visible) {
    return {
        type: STATE_DIALOG_VISIBLE_CHANGE,
        page: STATE_DIALOG_PAGE_SEARCH,
        visible
    };
}

/**
 * Handle dialog visibility state change.
 *
 * @function
 * @param {boolean} visible - The new visible state
 * @returns {object} The redux action playload
 */
function changeDialogVisibleTheme(visible) {
    return {
        type: STATE_DIALOG_VISIBLE_CHANGE,
        page: STATE_DIALOG_PAGE_THEME,
        visible
    };
}

export {
    changeDialogVisible,
    changeDialogVisibleBroadcast,
    changeDialogVisibleSearch,
    changeDialogVisibleTheme
};
