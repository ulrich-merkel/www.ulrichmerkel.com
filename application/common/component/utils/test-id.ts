/**
 * Es6 module for creating test id related attributes
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isValidString } from '../../utils/string';

type TestIdAttributes = {
    'data-testid': string;
};

/**
 * Get component properties for testing.
 *
 * @param {string} testId - The test id to be added
 * @returns {object|null} The component props for spreading
 */
export function getTestIdAttributes(testId?: string): TestIdAttributes | null {
    return isValidString(testId)
        ? {
              'data-testid': testId
          }
        : null;
}
