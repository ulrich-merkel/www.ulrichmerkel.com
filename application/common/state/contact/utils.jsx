/**
 * Es6 module for Redux Architecture.
 * Simple helper component to add utility functions.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2016
 * @version 0.0.1
 *
 * @see {@link http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html}
 *
 * @TODO Should be refactored and moved to actions
 *
 * @requires common/utils/validation
 *
 * @changelog
 * - 0.0.1 Basic functions and structure
 */
import { required, email } from '../../utils/validation';
import { isBrowser } from '../../utils/environment';

/**
 * Validation helper to check all inputs.
 *
 * @function
 * @param {object} [state={}] - The current redux contact state if given
 * @returns {object} The validation status for all inputs as object
 */
function validate(state = {}) {
    return {
        name: required(state.name),
        email: required(state.email) && email(state.email),
        subject: required(state.subject),
        message: required(state.message)
    };
}

/**
 * Validation single or all input element(s), depending on parameter input.
 *
 * @function
 * @param {object} state - The current redux contact state
 * @returns {boolean} The validation status
 */
function isValid(state) {
    const validated = validate(state);

    return !Object.keys(validated).some(function someKey(key) {
        return validated[key] !== undefined && !validated[key];
    });
}

function canSendForm(state) {
    const isStateValid = isValid(state);

    return isStateValid
        && state.pristine
        && !state.pending
        && isBrowser(state);
}

export {
    canSendForm,
    isValid,
    validate
};
