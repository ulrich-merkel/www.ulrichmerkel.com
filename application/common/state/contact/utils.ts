/**
 * Es6 module for Redux Architecture.
 * Simple helper component to add utility functions.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 *
 * @see {@link http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html}
 */
import { required, email } from '../../utils/validation';
import { isBrowser } from '../../utils/environment';
import { ContactFormType, ValidateType } from './types';

/**
 * Validation helper to check all inputs.
 *
 * @param {object} [contactForm] - The current redux contact state if given
 * @returns {object} The validation status for all inputs as object
 */
export function validate(contactForm: ContactFormType): ValidateType {
    return {
        name: required(contactForm?.name),
        email: required(contactForm?.email) && email(contactForm?.email),
        subject: required(contactForm?.subject),
        message: required(contactForm?.message)
    };
}

/**
 * Validation single or all input element(s), depending on parameter input.
 *
 * @param {object} [contactForm] - The current redux contact state
 * @returns {boolean} The validation status
 */
export function isValid(contactForm: ContactFormType): boolean {
    const validated = validate(contactForm);
console.log({validated, contactForm})
    return !Object.keys(validated).some(function someKey(key) {
        return validated[key] !== undefined && !validated[key];
    });
}

/**
 * Check if the form can be send.
 *
 * @param {object} [contactForm] - The current redux contact state
 * @returns {boolean} Whether the submit button is enabled or not
 */
export function canSendForm(contactForm: ContactFormType): boolean {
    const isStateValid = isValid(contactForm);

    return (
        isStateValid &&
        contactForm.pristine &&
        !contactForm.pending &&
        isBrowser(contactForm)
    );
}
