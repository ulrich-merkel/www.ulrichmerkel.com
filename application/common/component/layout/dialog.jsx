/**
 * Es6 module for React Component.
 * Layout components merge modules to bigger parts of the
 * overall page layout.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2016
 * @version 0.0.4
 *
 * @requires react
 * @requires react-redux
 * @requires classnames
 * @requires lodash
 * @requires common/state/selectors
 * @requires common/state/dialog/actions
 * @requires common/component/decorator/add-content
 * @requires common/utils/content
 * @requires common/utils/environment
 * @requires common/component/module/article
 * @requires common/component/module/list
 * @requires common/component/grid/spaced
 * @requires common/component/grid/row
 * @requires common/component/grid/col
 * @requires common/component/element/button
 *
 * @changelog
 * - 0.0.4 Add isBroadcast and isSearch to props
 * - 0.0.3 Moved to stateless function
 * - 0.0.2 Rewritten for es2015
 * - 0.0.1 Basic functions and structure
 */
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import classnames from 'classnames';
import { throttle, isEqual, get } from 'lodash';

import { selectStateDialogVisible, selectStateDialogPage } from './../../state/selectors';
import { changeDialogVisible } from './../../state/actions';
import addContent from './../decorator/add-content';
import { getContentSection } from './../../utils/content';
import { isBrowser } from './../../utils/environment';

import GridSpaced from './../grid/spaced';
import GridRow from './../grid/row';
import GridCol from './../grid/col';
import Button from './../element/button';

/**
 * Class representing a component.
 *
 * @class
 * @extends React.Component
 */
class LayoutDialog extends Component {

    /**
     * The actual class constructor.
     *
     * This is usally unnecessary if we don't perform any actions here,
     * because a default constructor will call super(props) for us.
     * We do this just because of completeness.
     *
     * @constructs
     * @param {Object} [props] - The initial class properties
     * @returns {void}
     */
    constructor(props) {
        super(props);

        /**
         * A bind call or arrow function in a JSX prop will create a brand new
         * function on every single render. This is bad for performance, as it
         * will result in the garbage collector being invoked way more than is necessary.
         *
         * Unfortunately React ES6 classes do not autobind their methods like components created
         * with the older React.createClass syntax. There are several approaches to binding methods
         * for ES6 classes. A basic approach is to just manually bind the methods in the constructor
         *
         * We also throttle the scroll function here to avoid unnecessary function calls while scrolling.
         *
         * @see {@link https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md}
         */
        this.onClose = this.onClose.bind(this);
        this.onKeyDown = throttle(this.onKeyDown.bind(this), 100);
    }

    /**
     * Invoked once, only on the client (not on the server),
     * immediately after the initial rendering occurs.
     *
     * @function
     * @returns {void}
     */
    componentDidMount() {
        if (isBrowser()) {
            window.addEventListener('keydown', this.onKeyDown, false);
        }
    }

    /**
     * ShouldComponentUpdate is triggered before the re-rendering process starts,
     * giving the developer the ability to short circuit this process.
     *
     * @function
     * @param {Object} nextProps - The news props to be rendered
     * @returns {boolean} Whether to force component update or not
     */
    shouldComponentUpdate(nextProps) {
        return !isEqual(this.props, nextProps);
    }

    /**
     * Invoked immediately before a component is unmounted from the DOM.
     *
     * @function
     * @returns {void}
     */
    componentWillUnmount() {
        if (isBrowser()) {
            window.removeEventListener('keydown', this.onKeyDown, false);
        }
    }

    /**
     * Esc key press.
     *
     * @function
     * @private
     * @param {Object} event - Synthetic react event
     * @returns {void}
     */
    onKeyDown(event) {
        if (event && event.keyCode === 27) {
            this.onClose();
        }
    }

    /**
     * Handle button click.
     *
     * @function
     * @private
     * @param {Object} event - Synthetic react event
     * @returns {void}
     */
    onClose(event) {
        event && event.preventDefault(); // eslint-disable-line no-unused-expressions
        this.props.handleChangeDialogVisible(false);
    }

    /**
     * The required render function to return a single react child element.
     *
     * @function
     * @returns {React.Element} React component markup
     */
    render() {
        const {
            dialogVisible,
            dialogPage,
            isBroadcast,
            isSearch,
            content,
            page,
            className,
            children
        } = this.props;

        if (!dialogVisible || !page || dialogPage !== page) {
            return null;
        }

        const contentSection = getContentSection(content);
        const contentSectionNav = contentSection('nav') || {};
        const composedClassName = classnames('l-dialog', className, {
            'l-dialog--broadcast': isBroadcast,
            'l-dialog--search': isSearch
        });

        return (
            <dialog className={composedClassName} role='presentation'>
                <div className='l-dialog__content'>

                    {children}

                    <GridSpaced>
                        <GridRow>
                            <GridCol>
                                <Button
                                    title={contentSectionNav.btnCloseTitle}
                                    onClick={this.onClose}
                                >
                                    {contentSectionNav.btnCloseLabel}
                                </Button>
                            </GridCol>
                        </GridRow>
                    </GridSpaced>

                    <Button
                        className='l-dialog__button--close c-font-icon--close'
                        classNameLabel='is-visually-hidden'
                        title={contentSectionNav.btnCloseTitle}
                        onClick={this.onClose}
                        isSmall
                    >
                        {contentSectionNav.btnCloseLabel}
                    </Button>

                </div>

                <div className='l-dialog__background' role='presentation' />

            </dialog>
        );
    }

}

/**
 * Validate props via React.PropTypes helpers.
 *
 * @static
 * @type {Object}
 * @property {Function} [handleChangeDialogVisible=Function.prototype] - Redux action callback to control dialog visibility
 * @property {boolean} [dialogVisible=false] - Redux state whether this dialog is visible or not
 * @property {Object} [content={}] - The component content config
 */
LayoutDialog.propTypes = {
    handleChangeDialogVisible: PropTypes.func,
    children: PropTypes.node, // eslint-disable-line react/require-default-props
    className: PropTypes.string, // eslint-disable-line react/require-default-props
    dialogPage: PropTypes.string,
    dialogVisible: PropTypes.bool,
    isBroadcast: PropTypes.bool,
    isSearch: PropTypes.bool,
    content: PropTypes.object,  // eslint-disable-line react/forbid-prop-types
    page: PropTypes.string
};

/**
 * Set defaults if props aren't available.
 *
 * @static
 * @type {Object}
 * @see LayoutDialog.propTypes
 */
LayoutDialog.defaultProps = {
    handleChangeDialogVisible: Function.prototype,
    dialogPage: '',
    dialogVisible: false,
    isBroadcast: false,
    isSearch: false,
    content: {},
    page: ''
};

/**
 * The component will subscribe to Redux store updates. Any time it updates,
 * mapStateToProps will be called, Its result must be a plain object,
 * and it will be merged into the componentâ€™s props.
 *
 * @function
 * @private
 * @param {Object.<*>} state - The redux store state
 * @param {Object.<*>} [ownProps] - The current component props
 * @returns {Object}
 */
function mapStateToProps(state, ownProps) {
    return {
        dialogVisible: selectStateDialogVisible(state) || get(ownProps, 'dialogVisible'),
        dialogPage: selectStateDialogPage(state) || get(ownProps, 'dialogPage')
    };
}

const LayoutDialogContainer = connect(
    mapStateToProps,
    {
        handleChangeDialogVisible: changeDialogVisible
    }
)(addContent('LayoutDialog')(LayoutDialog));

export default LayoutDialogContainer;
export {
    LayoutDialog
};
