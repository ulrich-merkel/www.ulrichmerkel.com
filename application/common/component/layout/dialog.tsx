/**
 * Es6 module for React Component.
 * Layout components merge modules to bigger parts of the
 * overall page layout.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { default as React, Component, ReactNode } from 'react';
import { connect } from 'react-redux';
import classnames from 'classnames';
import { throttle, get, noop } from 'lodash';

import {
    selectStateDialogContent,
    selectStateDialogVisible
} from '../../state/dialog/selector';
import { changeDialogVisible } from '../../state/dialog/duck';
import { addContent } from '../decorator/add-content';
import { getContentSection } from '../../utils/content';
import { Event, eventPreventDefault } from '../../utils/event';
import { isBrowser } from '../../utils/environment';
import { closeDialog, showDialog } from '../../../client/utils/dialog';

import { GridSpaced } from '../grid/spaced';
import { GridRow } from '../grid/row';
import { GridCol } from '../grid/col';
import { Button } from '../element/button';
import { View } from '../element/view';

type Props = {
    onChangeDialogVisible?: () => void;
    children?: ReactNode;
    className?: string;
    dialogPage?: string;
    dialogVisible?: boolean;
    content?: Record<string, unknown>;
    page?: string;
};

/**
 * Class representing a component.
 *
 * @augments React.Component
 */
export class LayoutDialog extends Component<Props> {
    /**
     * The actual class constructor.
     *
     * @constructs
     * @param {object} [props] - The initial class properties
     * @returns {void}
     */
    constructor(props: Props) {
        super(props);

        this.onKeyDown = throttle(this.onKeyDown.bind(this), 100);
    }

    /**
     * Invoked once, only on the client (not on the server),
     * immediately after the initial rendering occurs.
     *
     * @returns {void}
     */
    componentDidMount(): void {
        if (isBrowser()) {
            window.addEventListener('keydown', this.onKeyDown, false);
        }
    }

    /**
     * Invoked immediately before a component is unmounted from the DOM.
     *
     * @returns {void}
     */
    componentWillUnmount(): void {
        if (isBrowser()) {
            window.removeEventListener('keydown', this.onKeyDown, false);
        }
    }

    /**
     * Esc key press.
     *
     * @private
     * @param {object} event - Synthetic react event
     * @returns {void}
     */
    onKeyDown(event: KeyboardEvent): void {
        const { onChangeDialogVisible = noop } = this.props;

        if (event && event.keyCode === 27) {
            onChangeDialogVisible(false);
        }
    }

    /**
     * Handle button close.
     *
     * @private
     * @param {object} event - Synthetic react event
     * @returns {void}
     */
    onClose = (event: Event): void => {
        const { onChangeDialogVisible = noop } = this.props;

        eventPreventDefault(event);
        onChangeDialogVisible(false);
    };

    /**
     * The required render function to return a single react child element.
     *
     * @returns {ReactElement} React component markup
     */
    render(): ReactNode {
        const {
            children,
            className,
            content,
            dialogPage = '',
            dialogVisible = false,
            page = ''
        } = this.props;

        if (!dialogVisible || !page || dialogPage !== page) {
            closeDialog();
            return null;
        }

        showDialog();

        const contentSection = getContentSection(content);
        const contentSectionNav = contentSection('nav') || {};
        const composedClassName = classnames('l-dialog', className);

        return (
            <dialog className={composedClassName} role="presentation">
                <View className="l-dialog__content">
                    {children}

                    <GridSpaced>
                        <GridRow>
                            <GridCol>
                                <Button
                                    testId="button-close"
                                    title={contentSectionNav.btnCloseTitle}
                                    onClick={this.onClose}
                                >
                                    {contentSectionNav.btnCloseLabel}
                                </Button>
                            </GridCol>
                        </GridRow>
                    </GridSpaced>

                    <Button
                        className="l-dialog__button--close c-font-icon--close"
                        classNameLabel="is-visually-hidden"
                        testId="button-x"
                        title={contentSectionNav.btnCloseTitle}
                        onClick={this.onClose}
                    >
                        {contentSectionNav.btnCloseLabel}
                    </Button>
                </View>
            </dialog>
        );
    }
}

/**
 * The component will subscribe to Redux store updates. Any time it updates,
 * mapStateToProps will be called, Its result must be a plain object,
 * and it will be merged into the component’s props.
 *
 * @private
 * @param {object.<*>} state - The redux store state
 * @param {object.<*>} [ownProps] - The current component props
 * @returns {object}
 */
function mapStateToProps(state, ownProps) {
    return {
        dialogVisible:
            selectStateDialogVisible(state) || get(ownProps, 'dialogVisible'),
        dialogPage:
            selectStateDialogContent(state) || get(ownProps, 'dialogPage')
    };
}

/**
 * If an object is passed, each function inside it will be assumed to
 * be a Redux action creator. An object with the same function names,
 * but with every action creator wrapped into a dispatch call so they
 * may be invoked directly, will be merged into the component’s props.
 * If a function is passed, it will be given dispatch.
 *
 * @private
 * @type {object<string, Function}
 */
const mapDispatchToProps = {
    onChangeDialogVisible: changeDialogVisible
};

/**
 * Connects a React component to a Redux store. It does not modify the
 * component class passed to it. Instead, it returns a new, connected component class.
 *
 * @type {ReactElement}
 */
export const LayoutDialogConnected = connect(
    mapStateToProps,
    mapDispatchToProps
)(addContent('LayoutDialog')(LayoutDialog));
