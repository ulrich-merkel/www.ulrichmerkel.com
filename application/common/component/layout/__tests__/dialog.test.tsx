import * as React from 'react';
import { render } from '../../../../__tests__/utils/test-utils';
import { LayoutDialog, LayoutDialogConnected } from '../dialog';

describe('LayoutDialog', function fnDescribe() {
    const props = {
        changeDialogVisible: jest.fn(),
        className: 'test',
        dialogPage: 'foo',
        dialogVisible: true,
        content: {},
        page: 'foo'
    };

    it('should render correctly', function fnIt() {
        const { asFragment } = render(
            <LayoutDialog {...props}>Dialog Children</LayoutDialog>
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render correctly the connected version', function fnIt() {
        const { asFragment } = render(
            <LayoutDialogConnected {...props}>
                Dialog Children
            </LayoutDialogConnected>
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render null if not visible', function fnIt() {
        const { asFragment } = render(
            <LayoutDialog {...props} dialogVisible={false}>
                Dialog Children
            </LayoutDialog>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
