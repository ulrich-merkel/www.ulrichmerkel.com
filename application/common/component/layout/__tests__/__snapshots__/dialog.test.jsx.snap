// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutDialog should render correctly 1`] = `
<DocumentFragment>
  <dialog
    class="l-dialog"
    role="presentation"
  >
    <div
      class="l-dialog__content"
    >
      Dialog Children
      <div
        class="l-grid__spaced--small"
      >
        <div
          class="l-grid__row"
        >
          <div
            class="l-grid__col--12"
          >
            <button
              class="c-btn"
              role="button"
              title=""
              type="button"
            >
              <span
                class="c-btn__label"
              />
            </button>
          </div>
        </div>
      </div>
      <button
        class="c-btn l-dialog__button--close c-font-icon--close"
        role="button"
        title=""
        type="button"
      >
        <span
          class="c-btn__label is-visually-hidden"
        />
      </button>
    </div>
    <div
      class="l-dialog__background"
      role="presentation"
    />
  </dialog>
</DocumentFragment>
`;

exports[`LayoutDialog should render correctly the connected version 1`] = `
<DocumentFragment>
  <dialog
    class="l-dialog"
    role="presentation"
  >
    <div
      class="l-dialog__content"
    >
      Dialog Children
      <div
        class="l-grid__spaced--small"
      >
        <div
          class="l-grid__row"
        >
          <div
            class="l-grid__col--12"
          >
            <button
              class="c-btn"
              role="button"
              title=""
              type="button"
            >
              <span
                class="c-btn__label"
              />
            </button>
          </div>
        </div>
      </div>
      <button
        class="c-btn l-dialog__button--close c-font-icon--close"
        role="button"
        title=""
        type="button"
      >
        <span
          class="c-btn__label is-visually-hidden"
        />
      </button>
    </div>
    <div
      class="l-dialog__background"
      role="presentation"
    />
  </dialog>
</DocumentFragment>
`;

exports[`LayoutDialog should render null if not visible 1`] = `<DocumentFragment />`;
